File: piglatin.rb
class PigLatin

  VOWELS = ["A", "E", "I", "O", "U"]

  def self.translate(english)
    result = []
    result = english.split(/(\W)/).map do |word|
      word_translator(word)
    end
    result.join("")
  end

  def self.word_translator(word)
    word[0] == nil ? first_letter = "" : first_letter = word[0]
    if word == " " || word == "" || word == "?" || word == ","
      translated = word
    elsif VOWELS.include?(first_letter.upcase)
      translated = vowel_converter(word)
    else
      translated = consonant_converter(word)
    end
    translated
  end

  def self.consonant_converter(english)
    result = english.split("")
    while !VOWELS.include?(result.first.upcase)
      result.rotate!
    end
    result = result.join("") + "ay"
    result
  end

  def self.vowel_converter(english)
    english + "way"
  end

end

File: solution_in_js/piglatin.js
function translate(english) {
  result = []
  english.replace(/,/g, '').split(" ").forEach(function(word) {
    result.push(translator(word))
  })
  return result.join(" ")
};

function translator(word) {
  var vowels = ['a','e','i','o','u'];
  var leadingConsonants = []
  var translated = word

  if (vowels.includes(word[0].toLowerCase()) == true) {
    return word + "ay"
  }
  for (i=0; i<word.length; i++) {
    if (vowels.includes(word.charAt(i).toLowerCase())) {
      return (translated + leadingConsonants.join("") + "ay").toLowerCase();
    } else {
      leadingConsonants.push(word.charAt(i).toLowerCase())
      translated = word.substr(i+1)
    }
  }
}

console.log(translate("THrouGH"))
console.log(translate("through pig arrow"));
console.log(translate("through, pig, arrow"))
